include/envoy/ssl/connection.h

```diff
@@ -42,6 +44,11 @@ class Connection {
    *         certificate, or no SAN field, or no URI.
    **/
   virtual std::string uriSanPeerCertificate() PURE;
+
+  /**
+   * @return underlying SSL object for use with BoringSSL.
+   **/
+  virtual SSL* rawSslForTest() PURE;
```

Comment:
quick comment: Can we please not add this to the "public" interface. If you need to add to Ssl::ConnectionImpl and then do a dynamic_cast<> in the test that's OK.